lib/galaxy/tools/verify/__init__.py

```diff
@@ -97,6 +98,11 @@ def get_filename(filename):
         # if the server's env has GALAXY_TEST_SAVE, save the output file to that dir
         if keep_outputs_dir:
             ofn = os.path.join(keep_outputs_dir, filename)
+            out_dir = os.path.dirname(ofn)
+            try:
+                os.makedirs(out_dir)
+            except OSError as e:
+                log.debug('error creating {}, possibly because directory exists: {}'.format(out_dir, str(e)))
```

Comment:
What about doing `if not os.path.exists(out_dir)` instead of try-except and logging?

Reply:
Make sense - addressed in latest version of the patch.